RELEASE_GUIDE.md

```diff
@@ -21,6 +21,8 @@ make use of? If so:
 - Update node bindings
 - Update web bindings
 
+This can be verified by compiling [the latest package set](https://github.com/purescript/package-sets/releases/latest).
```

Comment:
I think I would rather put this as an item in the list directly under the "Before making a release" heading - "Check that there are no unintended breaking changes by compiling the latest package set." I think it's something which really ought to be done every time, whereas this frames it as a bit more of an optional suggestion.